docs/reference/fleet/fleet-search.asciidoc

```diff
@@ -0,0 +1,70 @@
+[role="xpack"]
+[[fleet-search]]
+=== Fleet search API
+++++
+<titleabbrev>Fleet search</titleabbrev>
+++++
+
+The purpose of the fleet search api is to provide a search api where the search
+will only be executed after provided checkpoint has been processed and is visible
+for searches inside of Elasticsearch.
+
+This API is experimental and only intended for internal use by fleet server.
+
+[discrete]
+[[wait-for-checkpoint-functionality]]
+== Wait for checkpoint functionality
+
+The fleet search API provides all of the expected functionality of the normal
+Elasticsearch search API. However, it also supports the optional parameter
+`wait_for_checkpoints`. This parameter is a list of sequence number checkpoints.
+When this parameter is present, the search will only be executed on local shards
+when the relevant sequence number checkpoint is visible for search after a refresh.
+The checkpoints are indexed by shard.
+
+If a timeout occurs before the the checkpoint has been refreshed into Elasticsearch,
+the search request will timeout.
+
+The fleet search API only supports searches against a single index.
+
+[discrete]
+[[fleet-search-partial-responses]]
+== Allow partial results
+
+By default, the Elasticsearch search api will allow <<search-partial-responses,partial search results>>.
+With this fleet API, it is common to configure this to be `false` or to check in the response
+to ensure each shard search was successful. If these precautions are not taken, it is
+possible for search results to be successfully returned even if a one or more shards
+timed out.
+
+[[fleet-search-api-request]]
+==== {api-request-title}
+
+`GET /<index>/_fleet/_search`
+
+[[fleet-search-api-path-params]]
+==== {api-path-parms-title}
+
+`<index>`::
+(Required, string)
+A single index or index alias that resolves to a single index.
```

Comment:
Do we need to add a check to ensure that we restrict to fleet indices or to indices with a single shard ?

Reply:
I don't think so. That is a requirement for the global checkpoint API they use. It would be impossible for their functionality to work with multiple shards.

Reply:
Also this doc was wrong. Aliases are not supported. I changed the doc.

Reply:
ok thanks